src/v/cluster/scheduling/partition_allocator.cc

```diff
@@ -240,18 +232,58 @@ std::error_code partition_allocator::check_cluster_limits(
     return errc::success;
 }
 
+ss::future<result<allocation_units::pointer>>
+partition_allocator::allocate(raw_allocation_request raw) {
+    vlog(
+      clusterlog.trace,
+      "allocation request for {} partitions",
+      raw.partition_count);
+
+    const uint64_t create_count = static_cast<uint64_t>(raw.partition_count)
+                                  * static_cast<uint64_t>(
+                                    raw.replication_factor);
+    auto cluster_errc = check_cluster_limits(create_count);
```

Comment:
Worth adding a comment that we must check limits before adding as to not materialize large vectors.